src/components/ReportActionItem/ReportPreview.tsx

```diff
@@ -136,6 +137,7 @@ function ReportPreview({
         [transactions, iouReportID, action],
     );
 
+    const [shouldStartPaidAnimation, setShouldStartPaidAnimation] = useState(false);
```

Comment:
Changes look good !
The only thing I would prefer
Is to pass only ref and use the state inside AnimatedSettlementButton since stopAnimation looks redundant

But it's optional

Reply:
I don't think using ref is a good idea since it won't trigger a re-render.